src/modules/Builder/features/ActivityItems/ActivityItems.tsx

```diff
@@ -53,6 +53,7 @@ export const ActivityItems = () => {
     itemToDelete,
     activity.conditionalLogic,
   );
+  const itemsWithVariablesToRemove = getItemsWithVariablesToRemove(itemToDelete?.name, items);
```

Comment:
may be it would be better
`const { string: ..., list: ... } = getItemsWithVariablesToRemove(...)`